Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectReplaceFitResult.py

```diff
@@ -14,46 +14,46 @@
     PropertyMode,
     WorkspaceGroup,
 )
+from typing import List, Union
 from mantid.kernel import Direction
 
 
-def string_ends_with(string, delimiter):
-    delimiter_length = len(delimiter)
-    return string[-delimiter_length:] == delimiter if len(string) > delimiter_length else False
+def string_ends_with(string: str, delimiters: List[str]) -> bool:
+    return any([string.endswith(delimiter) for delimiter in delimiters])
 
 
-def exists_in_ads(workspace_name):
+def exists_in_ads(workspace_name: str) -> bool:
     return AnalysisDataService.doesExist(workspace_name)
 
 
-def get_ads_workspace(workspace_name):
+def get_ads_workspace(workspace_name: str) -> bool:
```

Comment:
```suggestion
def get_ads_workspace(workspace_name: str) -> Union[None, MatrixWorkspace, WorkspaceGroup]:
```
Are these all the return types here?

Reply:
Yes